src/components/dashboard/GovernanceSection/GovernanceSection.tsx

```diff
@@ -30,10 +37,26 @@ const GovernanceSection = () => {
   const isDarkMode = useDarkMode()
   const theme = isDarkMode ? 'dark' : 'light'
   const { getAllowedFeaturesList } = useBrowserPermissions()
-  const [claimingSafeApp, errorFetchingClaimingSafeApp] = useRemoteSafeApps(SafeAppsTag.SAFE_CLAIMING_APP)
-  const claimingApp = claimingSafeApp?.[0]
+  const { safe, safeAddress } = useSafeInfo()
+  const granted = useIsGranted()
+  const chain = useCurrentChain()
+  const [matchingApps, errorFetchingClaimingSafeApp] = useRemoteSafeApps(SafeAppsTag.SAFE_CLAIMING_APP)
+  const claimingApp = matchingApps?.[0]
+  const { iframeRef, setAppIsLoading } = useAppIsLoading()
   const fetchingSafeClaimingApp = !claimingApp && !errorFetchingClaimingSafeApp
 
+  // Initialize the app communicator
+  useAppCommunicator(iframeRef, claimingApp, chain, {
+    onGetSafeInfo: () => ({
+      safeAddress,
+      chainId: chain ? parseInt(chain.chainId, 10) : -1,
+      owners: safe.owners.map((owner) => owner.value),
+      threshold: safe.threshold,
+      isReadOnly: !granted,
+      network: chain ? getLegacyChainName(chain.chainName || '', chain.chainId).toUpperCase() : '',
+    }),
```

Comment:
Perhaps we can extract the handlers so we can reuse it here?

Reply:
yes but this is like an instance of the climing app isn't it? so is a sdk method call we have to count

Reply:
Agree that tracking the safe info call is OK in this case.

> Perhaps we can extract the handlers so we can reuse it here?

That's a good idea, I'll see if it can be done.